src/components/Separator/Separator.tsx

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2023 New Vector Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import classnames from "classnames";
+import * as RadixSeparator from "@radix-ui/react-separator";
+import React, {
+  ComponentProps,
+  PropsWithoutRef,
+  Ref,
+  forwardRef,
+  useEffect,
+} from "react";
+import styles from "./Separator.module.css";
+
+type SeparatorProps = {
+  /**
+   * The CSS class name.
+   */
+  className?: string;
+  /**
+   * The spacing on either side of the separator
+   */
+  spacing?: string;
+} & ComponentProps<typeof RadixSeparator.Root>;
+
+const SPACING_CUSTOM_PROP = "--cpd-separator-spacing";
+
+export const Separator = forwardRef(
+  (
+    {
+      className,
+      spacing = "var(--cpd-space-2x)",
+      ...props
+    }: PropsWithoutRef<SeparatorProps>,
+    ref?: Ref<HTMLDivElement>,
+  ) => {
+    const classes = classnames(styles.separator, className);
+
+    useEffect(() => {
+      // Casting to `RefObject` as everything here should use this type.
+      // Functions refs are not something we wish to support and string refs
+      // are already unsupported with `forwardRef`.
```

Comment:
Please move this doc to component level so consumers can see this without needing to open the source code to avoid very difficult to diagnose bugs.

Reply:
I'd rather not overload the JSDoc comment. If that's a concern, i will instead throw an exception

Reply:
Throwing an exception is a runtime protection, we should be leveraging build time protections really

Reply:
How is that closer? it will still fail silently... i'd rather have a slightly delayed feedback but something a bit more explicit.

I guess i'll do both 👍

Reply:
Only ended up adding disclaimer in JSDoc to avoid extra complexity with throwing, this required some ninja moves in TypeScript that would the same complexity as the other solution (hook based)